src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.cs

```diff
@@ -20,39 +20,51 @@ namespace Uno.Toolkit.UI;
 /// <summary>
 /// Displays a view that replicates the look and behavior of the native splash screen
 /// </summary>
-[TemplatePart(Name = SplashScreenPresenterPartName, Type = typeof(ContentPresenter))]
 public partial class ExtendedSplashScreen : LoadingView
 {
-	private const string SplashScreenPresenterPartName = "SplashScreenPresenter";
-
 	public SplashScreen? SplashScreen { get; set; }
+	public Window? Window { get; set; }
+
+	#region DependencyProperty: SplashScreenContent
+
+	public static DependencyProperty SplashScreenContentProperty { get; } = DependencyProperty.Register(
+		nameof(SplashScreenContent),
+		typeof(object),
+		typeof(ExtendedSplashScreen),
+		new PropertyMetadata(default(object)));
+
+	/// <summary>
+	/// Gets or sets the native splash screen content to be displayed during loading/waiting.
+	/// </summary>
+	public object SplashScreenContent
+	{
+		get => (object)GetValue(SplashScreenContentProperty);
+		set => SetValue(SplashScreenContentProperty, value);
+	}
+
+	#endregion
 
 	protected override void OnApplyTemplate()
 	{
 		base.OnApplyTemplate();
 
-		if (SplashScreen is null)
-		{
-			this.Log().LogWarning("SplashScreen property has not been set (typically done in the OnLaunched Application method)");
-			return;
-		}
-
-		if (GetTemplateChild(SplashScreenPresenterPartName) is ContentPresenter splashScreenPresenter)
-		{
-			splashScreenPresenter.Content = GetNativeSplashScreen(SplashScreen);
-		}
-		else
-		{
-			this.Log().LogWarning($"Template for {nameof(ExtendedSplashScreen)} doesn't contain {nameof(ContentPresenter)} with Name set to {SplashScreenPresenterPartName}");
-		}
+		_ = LoadNativeSplashScreen();
+
+	}
+
+	private async Task LoadNativeSplashScreen()
+	{
+		var splashScreenContent = await GetNativeSplashScreen(SplashScreen);
 
+		// Return a non-visible element to make sure some content is set on the ContentPresenter
+		// Setting null on WinUI throws an exception
+		SplashScreenContent = splashScreenContent ?? new TextBlock { Text = "" };
```

Comment:
I'm curious what happens if we just don't set `SplashScreenContent` at all if `splashScreenContent` is null. Shouldn't the ContentPresenter already contain the ImplicitTextBlock in that case? I don't fully understand the ImplicitTextBlock stuff in ContentControl/Presenter though so this may not be relevant at all

Reply:
for a non-visible item, you can use a `<Border />` which is probably lighter than TextBlock; do note that as opposed to TextBlock, this by default is hv-stretched

Reply:
Ok will switch to Border.
@kazo0 yeh something weird on WinUI - seems to blow up when we try an animate the opacity to 0 where there's no Content set on the ContentPresenter